python/niizarr/_zarr2nii.py

```diff
@@ -1,33 +1,43 @@
+import argparse
 import io
 import sys
+import warnings
+
+import dask.array
+import numpy as np
 import zarr.hierarchy
 import zarr.storage
-import numpy as np
-import dask.array
-import argparse
 from nibabel import (Nifti1Image, Nifti1Header, Nifti2Image, Nifti2Header,
                      save, load)
-from ._header import HEADERTYPE1, HEADERTYPE2
+from nibabel.nifti1 import Nifti1Extension, Nifti1Extensions
+
+from ._header import bin2nii, NIFTI_1_HEADER_SIZE, NIFTI_2_HEADER_SIZE
 
 # If fsspec available, use fsspec
 try:
     import fsspec
+
     open = fsspec.open
 except (ImportError, ModuleNotFoundError):
     fsspec = None
 
 
-def bin2nii(buffer):
-    header = np.frombuffer(buffer, dtype=HEADERTYPE1, count=1)[0]
-    if header['magic'].decode() not in ('ni1', 'n+1', 'nz1'):
-        header = header.newbyteorder()
-    if header['magic'].decode() not in ('ni1', 'n+1', 'nz1'):
-        header = np.frombuffer(buffer, dtype=HEADERTYPE2, count=1)[0]
-        if header['magic'].decode() not in ('ni2', 'n+2', 'nz2'):
-            header = header.newbyteorder()
-        if header['magic'].decode() not in ('ni2', 'n+2', 'nz2'):
-            raise ValueError('Is this a nifti header?')
-    return header
+def extract_extension(chunk, index=0):
+    sections = []
+    chunk_len = len(chunk)
+
+    while index < chunk_len:
+        size = int.from_bytes(chunk[index:index + 4], byteorder='big')
```

Comment:
Why is big endian hardcoded here? Don't you want to assume the same endianness as guessed form the header?